src/app/rpmostree-ex-builtin-jigdo2commit.c

```diff
@@ -204,30 +211,110 @@ impl_jigdo2commit (RpmOstreeJigdo2CommitContext *self,
     oirpm_pkg = g_object_ref (pkglist->pdata[0]);
 
     /* Iterate over provides directly to provide a nicer error on mismatch */
-    gboolean found_v1_provide = FALSE;
+    gboolean found_vprovide = FALSE;
     g_autoptr(DnfReldepList) provides = dnf_package_get_provides (oirpm_pkg);
     const gint n_provides = dnf_reldep_list_count (provides);
     for (int i = 0; i < n_provides; i++)
       {
         DnfReldep *provide = dnf_reldep_list_index (provides, i);
 
-        if (g_str_equal (dnf_reldep_to_string (provide), RPMOSTREE_JIGDO_PROVIDE_V1))
+        const char *provide_str = dnf_reldep_to_string (provide);
+        if (g_str_equal (provide_str, RPMOSTREE_JIGDO_PROVIDE_V2))
+          {
+            found_vprovide = TRUE;
+          }
+        else if (g_str_has_prefix (provide_str, RPMOSTREE_JIGDO_PROVIDE_COMMIT))
           {
-            found_v1_provide = TRUE;
-            break;
+            const char *rest = provide_str + strlen (RPMOSTREE_JIGDO_PROVIDE_COMMIT);
+            if (*rest != '(')
+              return glnx_throw (error, "Invalid %s", provide_str);
+            rest++;
+            const char *closeparen = strchr (rest, ')');
+            if (!closeparen)
+              return glnx_throw (error, "Invalid %s", provide_str);
+
+            provided_commit = g_strndup (rest, closeparen - rest);
+            if (strlen (provided_commit) != OSTREE_SHA256_STRING_LEN)
+              return glnx_throw (error, "Invalid %s", provide_str);
           }
       }
 
-    if (!found_v1_provide)
+    if (!found_vprovide)
       return glnx_throw (error, "Package '%s' does not have Provides: %s",
```

Comment:
Not new, though let's give a more descriptive message here? E.g. something like "Package '%s' does not look like a jigdo RPM (missing Provides: %s)"  ?

Reply:
Let me punt this as it'd cause a chain of conflicts.  I'll be reworking all of this later; I'd like to put the "find jigdo RPM" query in `rpmostree-core.c`.